lib/poller.js

```diff
@@ -163,16 +163,60 @@ class Poller {
     }
 
     const secretManifest = await this._createSecretManifest()
-    this._logger.info(`upserting secret ${this._namespace}/${this._name}`)
+    const kubeSecret = kubeNamespace.secrets(this._name)
+    let existingSecret
 
     try {
-      return await kubeNamespace.secrets.post({ body: secretManifest })
+      this._logger.info(`getting secret ${this._namespace}/${this._name}`)
+      const secretResponse = await kubeSecret.get()
+      existingSecret = secretResponse.body
     } catch (err) {
-      if (err.statusCode !== 409) throw err
-      return kubeNamespace.secrets(this._name).put({ body: secretManifest })
+      if (err.statusCode !== 404) throw err
+      // do nothing if the secret is not found
+    }
+
+    if (existingSecret && this._equalSecretData(existingSecret, secretManifest)) {
+      this._logger.info(`skipping secret ${this._namespace}/${this._name} upsert, objects are the same`)
+      return Promise.resolve(true)
+    } else if (existingSecret === undefined) {
+      this._logger.info(`creating secret ${this._namespace}/${this._name}`)
+      return await kubeNamespace.secrets.post({ body: secretManifest })
+    } else {
+      this._logger.info(`updating secret ${this._namespace}/${this._name}`)
+      return kubeSecret.put({ body: secretManifest })
     }
   }
 
+  /**
+   * Checks if a secret and the desired secret manifest are equal
+   *
+   * @param   {Object}  kubeSecret      An actual kubernetes secret from the kube-client
+   * @param   {Object}  secretManifest  A hash representing the desired secret
+   *
+   * @return  {Boolean}                  Boolean if they are the same or not
+   */
+  _equalSecretData (kubeSecret, secretManifest) {
+    let result = true
+
+    // check custom metadata
+    const liveMetadata = clonedeep(kubeSecret.metadata)
+    const filterList = [
+      'creationTimestamp',
+      'namespace',
+      'ownerReferences',
+      'resourceVersion',
+      'selfLink',
+      'uid'
+    ]
+    filterList.forEach(k => delete liveMetadata[k])
```

Comment:
This `filterList` needs to be applied to the `kubeSecret` (the current data) aswell when comparing the stringified json.